lib/pure/nativesockets.nim

```diff
@@ -445,9 +445,10 @@ proc getSockDomain*(socket: SocketHandle): Domain =
   if getsockname(socket, cast[ptr SockAddr](addr(name)),
                  addr(namelen)) == -1'i32:
     raiseOSError(osLastError())
-  try:
+  let knownDomain = toKnownDomain(name.sin6_family.cint)
+  if knownDomain.isSome:
     result = toKnownDomain(name.sin6_family.cint).get()
```

Comment:
uhh, this is not correct since you call it again, you should have "knownDomain.get()" instead :P

Reply:
Yes. I'm very careless. :(

Reply:
It's fixed.